pkg/scale/varying_data_type_nested_example_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale_test
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// ParentVDT is a VaryingDataType that consists of multiple nested VaryingDataType
+// instances (aka. a rust enum containing multiple enum options)
+type ParentVDT scale.VaryingDataType
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (pvdt *ParentVDT) Set(val scale.VaryingDataTypeValue) (err error) {
+	// cast to VaryingDataType to use VaryingDataType.Set method
+	vdt := scale.VaryingDataType(*pvdt)
+	err = vdt.Set(val)
+	if err != nil {
+		return
+	}
+	// store original ParentVDT with VaryingDataType that has been set
+	*pvdt = ParentVDT(vdt)
+	return
+}
+
+// Value will return value from underying VaryingDataType
+func (pvdt *ParentVDT) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*pvdt)
+	return vdt.Value()
+}
+
+// NewParentVDT is constructor for ParentVDT
+func NewParentVDT() ParentVDT {
+	// use standard VaryingDataType constructor to construct a VaryingDataType
+	vdt, err := scale.NewVaryingDataType(NewChildVDT(), NewOtherChildVDT())
```

Comment:
Is it needed/is there a way to make the number of children here a param?

Reply:
The types of children and number of child options is supplied via the `NewVaryingDataType` constructor.  This is supposed to resemble the declaration of the enum in Rust.